src/services/uniswapV3/LiquidityService.ts

```diff
@@ -0,0 +1,415 @@
+import {Logger} from 'winston';
+import {inject, injectable} from 'inversify';
+import {Token, CurrencyAmount} from '@uniswap/sdk-core';
+import {FeeAmount, Tick, Pool, TICK_SPACINGS, TickMath, tickToPrice} from '@uniswap/v3-sdk';
+import axios from 'axios';
+import JSBI from 'jsbi';
+// eslint-disable-next-line max-len
+import IUniswapV3PoolABI from '@uniswap/v3-core/artifacts/contracts/interfaces/IUniswapV3Pool.sol/IUniswapV3Pool.json' assert {type: 'json'};
+import {ethers} from 'ethers';
+
+import {LeavesAndFeeds} from '../../types/Consensus.js';
+import {GraphTick, BarChartTick, LiquidityTick, TickProcessed, UniswapV3Param} from './interfaces.js';
+import {BlockchainProviderRepository} from '../../repositories/BlockchainProviderRepository.js';
+import {ChainsIds} from '../../types/ChainsIds.js';
+
+enum Direction {
+  ASC,
+  DESC,
+}
+
+const MAX_INT128 = JSBI.subtract(JSBI.exponentiate(JSBI.BigInt(2), JSBI.BigInt(128)), JSBI.BigInt(1));
+
+@injectable()
+export default class LiquidityService {
+  @inject('Logger') logger!: Logger;
+  @inject(BlockchainProviderRepository) blockchainProviderRepository!: BlockchainProviderRepository;
+
+  async getFullTickData(poolAddress: string): Promise<GraphTick[]> {
+    let allTicks: GraphTick[] = [];
+    let skip = 0;
+    let loadingTicks = true;
+    while (loadingTicks) {
+      const ticks = await this.getTickDataFromSubgraph(poolAddress, skip);
+      allTicks = allTicks.concat(ticks);
+      if (ticks.length < 1000) {
+        loadingTicks = false;
+      } else {
+        skip += 1000;
+      }
+    }
+
+    return allTicks;
+  }
+
+  async getTickDataFromSubgraph(poolAddress: string, skip: number): Promise<GraphTick[]> {
```

Comment:
how many calls we need to fetch ticks for one pool?

Reply:
ok, let's make sure number of ticks is configured with default value +-100

Reply:
But this one we need to get all Ticks from subgraph. Because based on that we now what is the current Tick and then we could evaluate the +- 100 ticks.
So I think we should use the maxLimit 1k here. Like, if we got the first 100 it doesn't mean we are getting the current Tick.
The +-100 is places on `src/services/dexes/uniswapV3/BarChartTicksGenerator.tsL45`

Let me know anything

Reply:
you right, this is wrong place. all good then.